### PR TITLE
Added the Tabs and Icons Component.

### PR DESCRIPTION
**Pull request by cyanff**
Changed credo config to ensure the following doesn't fail checks:
- todo tags
- moduledocs (we'll just add them on release, components are constantly in flux, it doesn't make sense to add module docs to everything right now)

### CODE REVIEW

File: apps/compounds_tester/lib/compounds_tester_web/components/compounds/icon.ex [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  slot :inner_block
+  attr :src, :string
+  attr :class, :string, default: nil
+
+  def icon(assigns) do
+    assigns = assign(assigns, :src_exists, assigns[:src] != nil)
+    assigns = assign(assigns, :inner_block_exists, not Enum.empty?(assigns.inner_block))
+
+    if xnor(assigns.src_exists, assigns.inner_block_exists) do
+      raise "You must provide EITHER an SVG element as a child to this component,
```
Comment (calebgasser) [inline apps/compounds_tester/lib/compounds_tester_web/components/compounds/icon.ex]: Should we handle this as a warning log? It begs the question if anything "breaking" in the components should handle it gracefully and log a warning or fail compilation.

File: apps/compounds_tester/lib/compounds_tester_web/components/compounds/tabs.ex [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  end
+
+  slot :inner_block
+  attr :id, :string, required: true
+  attr :disabled, :boolean, default: false
+
+  def tab(assigns) do
+    ~H"""
+    <button
+      id={"tab-"<>@id}
```
Comment (calebgasser) [inline apps/compounds_tester/lib/compounds_tester_web/components/compounds/tabs.ex]: We should use the auto generated id function as a default and allow them to pass in their own id.

File: apps/compounds_tester/lib/compounds_tester_web/components/compounds/tabs.ex [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  slot :inner_block
+  attr :class, :string, default: nil
+  attr :active_id, :string, default: nil
+
+  def tabs(assigns) do
+    assigns = assign(assigns, :tabs_id, "tabs-" <> Ecto.UUID.generate())
+
+    ~H"""
+    <div
+      id={@tabs_id}
```
Comment (calebgasser) [inline apps/compounds_tester/lib/compounds_tester_web/components/compounds/tabs.ex]: We should use the auto generated id function as a default and allow them to pass in their own id.

File: apps/compounds_tester/lib/compounds_tester_web/components/compounds/tabs.ex [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    </div>
+    """
+  end
+
+  slot :inner_block
+  attr :id, :string, required: true
+
+  def content(assigns) do
+    ~H"""
+    <div class="hidden w-full" id={"tab-content-" <> @id}>
```
Comment (calebgasser) [inline apps/compounds_tester/lib/compounds_tester_web/components/compounds/tabs.ex]: We should use the auto generated id function as a default and allow them to pass in their own id.

File: apps/compounds_tester/lib/compounds_tester_web/components/sandbox/tabs.ex
Comment (calebgasser) [inline apps/compounds_tester/lib/compounds_tester_web/components/sandbox/tabs.ex:1]: I don't know what the `sandbox` folder is for, just local testing? If so, probably shouldn't be merged in alpha. If there is some other purpose let me know.

**Comment by calebgasser** [comment]
We should also start writing tests. It's a drag, but the longer we wait to write them the more we're going to have to write when we do.

File: apps/compounds_tester/lib/compounds_tester_web/components/compounds/icon.ex (hunk 1 above)
Comment (cyanff) [reply apps/compounds_tester/lib/compounds_tester_web/components/compounds/icon.ex]: Hmm, yeah, I thought about it and it definitely shouldn't just break. I'll default it to the `src` and log some warnings if both are passed in.

File: apps/compounds_tester/lib/compounds_tester_web/components/compounds/tabs.ex (hunk 2 above)
Comment (cyanff) [reply apps/compounds_tester/lib/compounds_tester_web/components/compounds/tabs.ex]: Yeah that's reasonable.

File: apps/compounds_tester/lib/compounds_tester_web/components/sandbox/tabs.ex
Comment (cyanff) [reply apps/compounds_tester/lib/compounds_tester_web/components/sandbox/tabs.ex:1]: Will remove.

**Comment by cyanff** [comment]
I don't think we should start writing tests before 1.0 alpha.
At this stage of the project, components evolve frequently as we learn more about what works and what doesn't.
It would be wasteful to enter of cycle of write component -> write test -> overhaul component -> scrap and rewrite new tests.
After having a draft of all components for 1.0 alpha, we could test the interactions of all the components when used together, which is much more valuable than individual unit tests.

**Comment by calebgasser** [comment]
> I don't think we should start writing tests before 1.0 alpha. At this stage of the project, components evolve frequently as we learn more about what works and what doesn't. It would be wasteful to enter of cycle of write component -> write test -> overhaul component -> scrap and rewrite new tests. After having a draft of all components for 1.0 alpha, we could test the interactions of all the components when used together, which is much more valuable than individual unit tests.

That makes sense. It should defiantly be a priority once we hit 1.0 alpha

**Comment by cyanff** [comment]
Instead of using ids to link `tab` and `content` I switched to using an attribute called `key`. 
This is so that users can pass in their own ids without breaking the component's interactivity.

**Review by calebgasser — APPROVED** [review]

**Review by che-122 — APPROVED** [review]

**Pull request merged**